source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeRepositoryTests.cs

```
@@ -66,7 +66,7 @@ public async Task AddAsync_WhenChargeIsValid_ChargeIsAdded()
             await using var chargesDatabaseReadContext = _databaseManager.CreateDbContext();
 
             var actual = await chargesDatabaseReadContext.Charges
-                .SingleOrDefaultAsync(x =>
+                .SingleAsync(x =>
```

This changes behavior, is this intended

---

Good question, as the change is unrelated to the .net and function version upgrade. In a test, we can presume to always get the expected data, and if not it is nice to fail fast. I think we should keep the adjustment.

---

I changed this because the analyzer complained that "actual" could be null, so as Henrik wrote, I felt it was better to fail fast than to use "actual!" in the code.